packages/components/src/components/Trigger/Trigger.stories.js

```diff
@@ -0,0 +1,145 @@
+/*
+Copyright 2019 The Tekton Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import React from 'react';
+import { storiesOf } from '@storybook/react';
+
+import { Router } from 'react-router-dom';
+import { IntlProvider } from 'react-intl';
+import { createMemoryHistory } from 'history';
+import Trigger from './Trigger';
+
+const route = '/';
+const history = createMemoryHistory({ initialEntries: [route] });
+const props = {
+  eventListenerNamespace: 'default',
+  trigger: {
+    name: 'my-trigger',
+    bindings: [
+      { name: 'triggerbinding0' },
+      { name: 'triggerbinding1' },
+      { name: 'triggerbinding2' }
+    ],
+    template: {
+      name: 'triggertemplate'
+    },
+    interceptors: [
+      {
+        webhook: {
+          header: [
+            {
+              name: 'header0',
+              value: 'value0'
+            },
+            {
+              name: 'header1',
+              value: ['value1-0', 'value1-1', 'value1-2']
+            }
+          ],
+          objectRef: {
+            apiVersion: 'v1',
+            kind: 'Service',
+            name: 'interceptor-service0',
+            namespace: 'foo'
+          }
+        }
+      },
+      {
+        github: {
+          secretRef: {
+            secretName: 'github-secret',
+            secretKey: 'secret'
+          },
+          eventTypes: ['push', 'pull_request']
+        }
+      },
+      {
+        gitlab: {
+          secretRef: {
+            secretName: 'gitlab-secret',
+            secretKey: 'secret',
+            namespace: 'foo'
+          },
+          eventTypes: ['Push Hook']
+        }
+      },
+      {
+        cel: {
+          filter: "body.matches('foo', 'bar')"
+        }
+      }
+    ]
+  }
+};
+
+storiesOf('Trigger', module)
+  .add('default', () => (
+    <Router history={history}>
+      <IntlProvider locale="en" defaultLocale="en" messages={{}}>
+        <Trigger {...props} />
+      </IntlProvider>
+    </Router>
+  ))
+  .add('no name', () => (
+    <Router history={history}>
+      <IntlProvider locale="en" defaultLocale="en" messages={{}}>
+        <Trigger
+          {...{ ...props, trigger: { ...props.trigger, name: undefined } }}
```

Comment:
The extra spread operator `{... }` makes this look odd and could cause confusion for others reading it. Maybe we could split the props to avoid it?

`<Trigger {...props} trigger={{ ...props.trigger, name: undefined }} />`

What do you think?

Reply:
Ok, yeah, that looks better 👍